controllers/harborconfiguration_controller.go

```diff
@@ -0,0 +1,421 @@
+
+	if err != nil && errors.Is(err, hErr) {
+		err = client.NewReplicationPolicy(ctx,
+			reqDestinationRegistry,
+			srcRegistry,
+			harborConfiguration.Spec.Replication.ReplicateDeletion,
+			harborConfiguration.Spec.Replication.Override,
+			harborConfiguration.Spec.Replication.EnablePolicy,
+			reqFilters,
+			reqTrigger,
+			harborConfiguration.Spec.Replication.DestinationNamespace,
+			harborConfiguration.Spec.Replication.Description,
+			harborConfiguration.Spec.Replication.Name)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+	} else if err == nil {
+		update := modelv2.ReplicationPolicy{
+			Name:          harborConfiguration.Spec.Replication.Name,
+			Description:   harborConfiguration.Spec.Replication.Description,
+			SrcRegistry:   srcRegistry,
+			DestNamespace: harborConfiguration.Spec.Replication.DestinationNamespace,
+			DestRegistry:  reqDestinationRegistry,
+			Filters:       reqFilters,
+			Trigger:       reqTrigger,
+			Override:      harborConfiguration.Spec.Replication.Override,
+			Enabled:       harborConfiguration.Spec.Replication.EnablePolicy,
+		}
+
+		err = client.UpdateReplicationPolicy(ctx, &update, replicationFound.ID)
+		// Experienced some flakiness
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+	} else {
+		return ctrl.Result{}, err
+	}
+	return ctrl.Result{}, err
+}
+
+func (r *HarborConfigurationReconciler) reconcileAll(ctx context.Context, harborConfiguration harborconfigurationv1alpha1.HarborConfiguration, client *apiv2.RESTClient) (ctrl.Result, error) {
+	registry := &modelv2.Registry{
+		Name:        harborConfiguration.Spec.Registry.Name,
+		Type:        harborConfiguration.Spec.Registry.Type,
+		URL:         harborConfiguration.Spec.Registry.TargetRegistryUrl,
+		Description: harborConfiguration.Spec.Registry.Description,
+		Credential:  (*modelv2.RegistryCredential)(harborConfiguration.Spec.Registry.Credential),
+	}
+
+	_, err := r.registryReconciliation(ctx, harborConfiguration, *registry, client)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	_, err = r.projectReconciliation(ctx, harborConfiguration, *registry, client)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	_, err = r.replicationRuleReconciliation(ctx, harborConfiguration, *registry, client)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	return ctrl.Result{}, err
+}
+
+func getHarborSecret(ctx context.Context, clientSet *kubernetes.Clientset, harborcluster *harborOperator.HarborCluster) (string, error) {
+	passwordSecret, err := clientSet.CoreV1().Secrets(harborcluster.Namespace).Get(ctx, harborcluster.Spec.HarborAdminPasswordRef, v1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+	return string(passwordSecret.Data["secret"]), nil
+}
+
+func getHarborURL(harborcluster *harborOperator.HarborCluster) string {
+	url := fmt.Sprintf("http://%s-harbor-harbor-core.%s/api/v2.0", harborcluster.Name, harborcluster.Namespace)
```

Comment:
Does `-harbor-harbor-core` part of the URL depend on how harbor app is deployed? Like `namespace` or `release name`?

Reply:
Yes, the `-harbor-harbor-core` suffix is a default from the harbor operator